### PR TITLE
Backend bug cleansing [PAR-73]

### PR DESCRIPTION
**Pull request by Hashiphoto**
Fixes:
Inject owner and requester objects into book
Return empty array when library is empty instead of null

### CODE REVIEW

**Comment by Hashiphoto** [comment]
I'm going to leave this open for now while we investigate more bugs

**Comment by Hashiphoto** [comment]
I think these are the only bugs that came up today. Go ahead and merge if approved

File: db.js [hunk 1]
```diff
@@ -43,6 +43,16 @@ var books = (function () {
       return rows[0].count;
     });
 
+    if (mostRecentLoan) {
+      var owner = await users.getById(book.user_id, currentUserId);
+      var requester = await users.getById(
```
Comment (lemniscate8) [inline db.js:48]: Is it guaranteed that the requester of this loan is the current user?

Comment (Hashiphoto) [reply db.js:48]: Maybe the parameter ordering is confusing, but the signature is `users.getById(currentUserId, targetUserId)`. In short, it's like, here's who I am, now who is this person? (because the status is contextual based on who you are). 

Tldr, the user retrieved here is the `mostRecentLoan.requester_id`
